exchange/auction_test.go

```diff
@@ -124,62 +142,80 @@ func runCacheSpec(t *testing.T, fileDisplayName string, specData *cacheSpec, bid
 	ctx := context.Background()
 	cache := &mockCache{}
 
+	targData := &targetData{
+		priceGranularity: openrtb_ext.PriceGranularity{
+			Precision: 2,
+			Ranges: []openrtb_ext.GranularityRange{
+				{
+					Min:       0,
+					Max:       5,
+					Increment: 0.05,
+				},
+				{
+					Min:       5,
+					Max:       10,
+					Increment: 0.1,
+				},
+				{
+					Min:       10,
+					Max:       20,
+					Increment: 0.5,
+				},
+			},
+		},
+		includeWinners:    specData.TargetDataIncludeWinners,
+		includeBidderKeys: specData.TargetDataIncludeBidderKeys,
+		includeCacheBids:  specData.TargetDataIncludeCacheBids || bids,
+		includeCacheVast:  specData.TargetDataIncludeCacheVast || vast,
+	}
+
 	testAuction := &auction{
 		winningBids:         winningBids,
 		winningBidsByBidder: winningBidsByBidder,
 		roundedPrices:       roundedPrices,
 	}
-	_ = testAuction.doCache(ctx, cache, bids, vast, &specData.BidRequest, 60, &specData.DefaultTTLs, bidCategory)
-	found := 0
+	_ = testAuction.doCache(ctx, cache, targData, &specData.BidRequest, 60, &specData.DefaultTTLs, bidCategory)
 
-	for _, cExpected := range specData.ExpectedCacheables {
-		for _, cFound := range cache.items {
-			// make sure Data section matches exactly
-			if !vast {
-				// not testing VAST XML for custom cache key test
-				eq := jsonpatch.Equal(cExpected.Data, cFound.Data)
-				if !eq {
-					continue
-				}
-			}
+	if !vast {
+		if len(specData.ExpectedCacheables) == len(cache.items) {
+			var ht map[string]int = make(map[string]int)
+			var compareString string
 
-			// make sure Key value is as expected
-			keymatch := false
-			if len(cExpected.Key) > 0 {
-				// can only verify prefix; remainder is random
-				if strings.HasPrefix(cFound.Key, cExpected.Key) {
-					keymatch = true
-				}
-			} else {
-				if len(cFound.Key) == 0 {
-					// Key is expected to be empty
-					keymatch = true
-				}
+			for _, cExpected := range specData.ExpectedCacheables {
+				compareString = string(cExpected.Data) + string(cExpected.TTLSeconds)
```

Comment:
I see you are just checking `Data`  and `TTLSeconds`. `Type` is not likely to be an issue. This does miss out on checking that custom keys are being set. You were not able to get the mock cache client to unrandomize the uuid in the custom key?

Reply:
I did not Hans. I didn't get the mock cache client to unrandomize the uuid in the custom key. Is it possible for the `gofrs/uuid` library to not deliver randomized UUID codes? Is this what your mean?

Reply:
No, I was thinking it might be possible for the mock client to do some pattern matching to recognize the uuid part and replace it. But I had not looked into it too closely.